legacy/application/forms/GeneralPreferences.php

```diff
@@ -207,6 +207,22 @@ public function init()
         $radioPageLoginButton->setValue($displayRadioPageLoginButtonValue);
         $this->addElement($radioPageLoginButton);
 
+        // add a checkbox for completely disabling the radio page
+        $radioPageDisabled = new Zend_Form_Element_Checkbox('radioPageDisabled');
+        $radioPageDisabled->setDecorators([
+            'ViewHelper',
+            'Errors',
+            'Label',
+        ]);
+        $radioPageDisabledValue = Application_Model_Preference::getRadioPageDisabled();
+        if ($radioPageDisabledValue == '') {
+            $radioPageDisabledValue = false;
+        }
```

Comment:
Please move this code the `Application_Model_Preference::getRadioPageDisabled()` function.